src/api/nlopt.h

```diff
@@ -301,12 +301,26 @@ NLOPT_EXTERN(nlopt_result) nlopt_minimize(nlopt_algorithm algorithm, int n, nlop
                                           double *minf, /* out: minimum */
                                           double minf_max, double ftol_rel, double ftol_abs, double xtol_rel, const double *xtol_abs, int maxeval, double maxtime) NLOPT_DEPRECATED;
 
+/* this is still used internally by a few algorithms, it's the same as nlopt_minimize, but without the deprecation */
+NLOPT_EXTERN(nlopt_result) internal_nlopt_minimize(nlopt_algorithm algorithm, int n, nlopt_func_old f, void *f_data,
+                                          const double *lb, const double *ub, /* bounds */
+                                          double *x,    /* in: initial guess, out: minimizer */
+                                          double *minf, /* out: minimum */
+                                          double minf_max, double ftol_rel, double ftol_abs, double xtol_rel, const double *xtol_abs, int maxeval, double maxtime);
+
```

Comment:
I think it would be better to just update the algorithms to use the new API.

(We shouldn't export an `internal_*` function in `nlopt.h` in any case.)

Reply:
Updated to eliminate these calls.